### PR TITLE
Add pod security policies for prometheus components

### PR DESCRIPTION
**Pull request by wfernandes**
#### What this PR does / why we need it:
This PR adds basic Pod Security Policies for the promethues components. This will allow this chart to be installed in an environment that has the PSP admission controller enabled. 

Interestingly, there was a PSP included just for the node-exporter job. We just mimicked that PSP and added some for the other jobs.

#### Special notes for your reviewer:
This is a duplicate of https://github.com/helm/charts/pull/12825 which as already been approved. Unfortunately, it got stale and the bot closed the PR. 

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by wfernandes** [comment]
@gianrubio @mgoodness As mentioned in the description, this PR is a duplicate of https://github.com/helm/charts/pull/12825 which was approved but it just had a merge conflict which I believe has been resolved.

**Comment by wfernandes** [comment]
@gianrubio @mgoodness As mentioned in the description, this PR is a duplicate of #12825 which was approved but it just had a merge conflict which I believe has been resolved.

File: stable/prometheus/values.yaml [hunk 1]
```diff
@@ -1,5 +1,7 @@
 rbac:
   create: true
+  podSecurityPolicy:
```
Comment (paulczar) [inline stable/prometheus/values.yaml]: I'm not sure enabling PSP is an RBAC related thing, so it probably doesn't belong inside the `rbac` structure here?

File: stable/prometheus/Chart.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: prometheus
-version: 8.14.3
+version: 8.14.4
```
Comment (paulczar) [inline stable/prometheus/Chart.yaml]: It's adding a new feature, so i would bump version to `8.15.0`

**Review by paulczar — CHANGES_REQUESTED** [review]

**Comment by Benjamintf1** [comment]
We pulled PSP enablement outside of rbac, and upped to 9.0.0 (because moving psp property from under node-exporter to a global property is a breaking change).

File: stable/prometheus/templates/node-exporter-role.yaml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 {{- if and .Values.nodeExporter.enabled .Values.rbac.create }}
-{{- if .Values.nodeExporter.podSecurityPolicy.enabled }}
+{{- if .Values.podSecurityPolicy.enabled }}
```
Comment (paulczar) [inline stable/prometheus/templates/node-exporter-role.yaml]: what if we did `{{- if or (default .Values.nodeExporter.podSecurityPolicy.enabled false) (.Values.podSecurityPolicy.enabled) }}` for these checks, and then put a deprecation message in the NOTES.txt ? 

that way its not yet a breaking change, but is also preparing people for deprecation of the old value ?

Comment (paulczar) [reply stable/prometheus/templates/node-exporter-role.yaml]: or if we want the breaking change we could error if `.Values.nodeExporter.podSecurityPolicy.enabled` is set with a error message with upgrade instructions.

Comment (Benjamintf1) [reply stable/prometheus/templates/node-exporter-role.yaml]: done!

**Comment by nico-ulbricht** [comment]
This PR seems to have introduced a bug whereas the following error occurs if `rbac.enabled` is true but `podSecurity.enabled` is false:

```
Error: rpc error: code = Unknown desc = validation failed: error validating "": error validating data: ValidationError(ClusterRole): missing required field "rules" in io.k8s.api.rbac.v1beta1.ClusterRole
```

Which seems to be caused by creating a ClusterRole without rules here: https://github.com/helm/charts/pull/15055/files#diff-055e5eaaea98daeb77e0619ee5effa11R9

**Comment by Benjamintf1** [comment]
@nico-ulbricht ahh, that makes sense. It should probubly encompase the entire clusterrole.

**Pull request merged**